client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -192,12 +206,28 @@ private boolean sendShuffleDataAsync(
     return result;
   }
 
-  private void genServerToBlocks(ShuffleBlockInfo sbi, List<ShuffleServerInfo> serverList,
+  void genServerToBlocks(
+      ShuffleBlockInfo sbi,
+      List<ShuffleServerInfo> serverList,
+      int replicaNum,
+      List<ShuffleServerInfo> excludeServers,
       Map<ShuffleServerInfo, Map<Integer, Map<Integer, List<ShuffleBlockInfo>>>> serverToBlocks,
-      Map<ShuffleServerInfo, List<Long>> serverToBlockIds) {
+      Map<ShuffleServerInfo, List<Long>> serverToBlockIds,
+      boolean withoutBlocklist) {
+    if (replicaNum <= 0) {
+      return;
+    }
     int partitionId = sbi.getPartitionId();
     int shuffleId = sbi.getShuffleId();
+    int assignedNum = 0;
     for (ShuffleServerInfo ssi : serverList) {
+      if (withoutBlocklist && replica > 1
+          && shuffleServerBlocklist.contains(ssi)) {
```

Comment:
This condition looks strange to me: `withoutBlocklist && shuffleServerBlocklist.contains(ssi)`.
Could you explain what `withoutBlocklist` is controlling here?
Maybe the name of this parameter still needs to be changed.

Reply:
if replicaNum > writeReplicaNum, the blocks may will be sended for two rounds. And `shuffleServerBlocklist` need to be ignored when sending the second round or servers will not be enough.

https://github.com/apache/incubator-uniffle/blob/b80972c226a3afd5568555a9a76fe3bb70ed3d9b/client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java#L249-L254

Reply:
My fault. Change to `withBlocklist`